### PR TITLE
Adding initial support for 20.04

### PR DESCRIPTION
**Pull request by Neotriple**
[//]: # "Thanks for your contribution.  To make life of the reviewers easier,"
[//]: # "please give this pull request a meaningful title and provide the"
[//]: # "requested information below."

## Description

This PR is initial support for 20.04 machines (not including RT-PREEMPT). The main things I have addressed so far is that:

- 20.04 does not support dashing or melodic, and thus needed to be updated to foxy and noetic. 
- 20.04 comes with Python3.8 by default, and thus, the LAAS package needed to download the appropriate versions of the software

There is definitely some further cleanup that needs to be done on the 20.04 scripts. 

[//]: # "Description of what you did.  If it is more complex, consider to add a"
[//]: # "'Summary' section."

I modified the setup bash files and generate_bash_script_from_dockerfile.py to accept 20.04 as an argument

## How I Tested

I have only done preliminary tests, this was in order to use dynamic_graph_head, the vicon SDK (private), and LAAS software as well as ros2. (i.e. I was successfully able to use and install these certain packages).

[//]: # "Explain how you tested your changes"

I have only tested using a real-time script used by someone in the Machines in Motion lab and was able to successfully run this.

## Do not merge before

[//]: # "Link other open pull request here that need to be merged first."
[//]: # "Remove this section if it does not apply."

- [ ] link to dependent pull request

## I fulfilled the following requirements

[//]: # "Please make sure you followed these steps before requesting a review."
[//]: # "Check the boxes in the list below, when done."

- [ ] All new code is formatted according to our style guide (for C++ run clang-format, for Python, run flake8 and fix all warnings).
- [ ] All new functions/classes are documented and existing documentation is updated according to changes.
- [ ] No commented code from testing/debugging is kept (unless there is a good reason to keep it).

### CODE REVIEW

File: official/ubuntu_20_04/docker/ros/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+  libasio-dev \
+  libtinyxml2-dev
+# install Cyclone DDS dependencies
+RUN apt install --no-install-recommends -y \
+  libcunit1-dev
+
+#
+# Dashing
+#
+RUN apt install -y ros-foxy-desktop `# All ROS2 base code.` \
```
Comment (MaximilienNaveau) [inline official/ubuntu_20_04/docker/ros/Dockerfile:121]: For ubuntu 20.04 the last version of ROS 2 is galactic, https://docs.ros.org/en/galactic/Installation/Ubuntu-Install-Debians.html
Would it make sens to use this or Foxy is better.
I am not sure which one is supported the longest.

**Review by MaximilienNaveau — APPROVED** [review]
Thanks for this!

File: official/ubuntu_20_04/docker/ros/Dockerfile (hunk 1 above)
Comment (MaximilienNaveau) [reply official/ubuntu_20_04/docker/ros/Dockerfile:121]: Ok never mind Foxy is a the the ROS2 LTS on ubunut20.04:
https://docs.ros.org/en/galactic/Releases.html#releases

**Comment by jviereck** [comment]
Getting rt-preempt to work was not as out of the box as hoped. By default, Ubuntu 20.04 is using an nvidia driver, which is not realtime compatible. To get things working I did:

- Edit `sudo vim /lib/modprobe.d/nvidia-graphics-drivers.conf` and comment out the blacklisting for the nouveau driver
- Change the boot options for grub "modprobe.blacklist=nvidia systemd.setenv=GPUMOD=nouveau" using `gedit /etc/default/grub` and `sudo update-grub`.
- Boot the rt-preempt kernel
- The resolution won't be good
- Run `$ sudo modprobe nouveau` - the screen will go black
- Switch between terminals using Ctrl+Alt+F1/F2/F3
- The screen will show-up at high resolution

It would be great if the modprobe command could be executed at startup. I tried the solution from [1] but it didn't work. I tried to load the module at startup but it didn't work.

[1]: https://askubuntu.com/questions/1339726/set-graphics-driver-to-nouveau-for-specific-grub-menuentry

**Comment by jviereck** [comment]
Doing a fresh install and running the `setup_ubuntu install all` script, the following packages are missing and required manual installation. They are part of the installation procedure, so I don't understand why they are missing.

The packages are reported by @Neotriple:

```
pip3 install breathe
sudo apt install libcereal-dev libedit-dev doxygen
```

@MaximilienNaveau , @vincentberenz  do you have any idea why the packages would be missing during the install procedure?

**Comment by Neotriple** [comment]
Did you also run into the issues regarding these passages @jviereck ? If you didn't, it could have been something on my end (or perhaps for 20.04 these packages had other dependencies which were not installed before they were called in the core script?)

**Comment by jviereck** [comment]
> Did you also run into the issues regarding these passages @jviereck 

Yes, I did. 

I don't think this should be a dependency issue. If there were dependent packages, they should just get installed as well.

**Comment by MaximilienNaveau** [comment]
I am merging this can you @Neotriple or @jviereck  can you create an issue here to explain again the problem with the not installed packages?

**Comment by jviereck** [comment]
Created a followup-issue in #27 .

**Pull request merged**